File: Codes in C++/Array/Container with most water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0;
        int n = height.size();
        int left = 0;
        int right = n - 1;
        int maxArea = 0;
        while (left < right) {
            int currArea = min(height[left], height[right]) * (right - left);
            maxArea = max(maxArea, currArea);
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return maxArea;
    }
};

File: Codes in Python3/Array/Container With Most Water.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        n=len(heights)
        l,r=0,n-1
        maxArea=0
        while l<r:
            area=min(heights[l],heights[r])*(r-l)
            maxArea= area if area>maxArea else maxArea
            if heights[l]<heights[r]:
                l+=1
            else:
                r-=1
            
        return maxArea

# TC=O(n)
# SC=O(1)
        